src/client/src/rt-components/tear-off/ExternalWindow.tsx

```diff
@@ -12,51 +11,46 @@ const defaultConfig: WindowConfig = {
 }
 
 export interface ExternalWindowProps {
-  title: string
-  onBlock: () => void
+  title?: string
+  onBlock?: () => void
   onUnload: () => void
-  config: WindowConfig
+  config?: WindowConfig
 }
 
-const defaultWindowProps: ExternalWindowProps = {
-  title: '',
-  onBlock: null as () => void,
-  onUnload: null as () => void,
-  config: defaultConfig,
-}
-
-class ExternalWindow extends React.Component<ExternalWindowProps & { platform: PlatformAdapter }> {
-  externalWindow: Window | null = null
-
-  async componentDidMount() {
-    const { config, platform } = this.props
+const ExternalWindow: FC<ExternalWindowProps> = ({
+  title = '',
+  onBlock = null as () => void,
+  onUnload = null as () => void,
+  config = defaultConfig,
+}) => {
+  const platform = usePlatform()
 
-    this.externalWindow = await platform.window.open(config, this.release)
+  let externalWindow: Window
 
-    if (this.externalWindow) {
-      window.addEventListener('beforeunload', this.release)
+  const release = () => {
+    if (externalWindow) {
+      window.removeEventListener('beforeunload', release)
     }
+    onUnload.call(null)
```

Comment:
Default value for onUnload is null but we are calling it here without null checking.

Reply:
The TS definitions require a null here. The default its `{}`

Reply:
I mean the `onUnload` prop can be null, and is in fact the default:
```
const ExternalWindow: FC<ExternalWindowProps> = ({
  onUnload = null as () => void,
```
Note that this problem was already there before this PR, just that the only place in which the component is used is providing a valid function so it hasn't surfaced before. I'd suggest something along the lines of `onUnload && onUnload.call(null)` or `typeof onUnload === 'function'` or similar